File: backend_python/app/models/process.py
from sqlalchemy import Column, Integer, String, Date, DateTime
from app.models.base import Base
from sqlalchemy.dialects.mysql import LONGTEXT
from datetime import datetime

class Process(Base):
    __tablename__ = "processes"
    id = Column(Integer, primary_key=True, index=True)
    numero_processo = Column(String(50), nullable=True)
    data_disponibilizacao = Column(Date, nullable=True)
    autores = Column(String(255), nullable=True)
    advogados = Column(String(255), nullable=True)
    conteudo_publicacao = Column(LONGTEXT, nullable=True)
    valor_bruto = Column(String(50), nullable=True)
    valor_liquido = Column(String(50), nullable=True)
    juros_moratorios = Column(String(50), nullable=True)
    honorarios_advocaticios = Column(String(50), nullable=True)
    reu = Column(String(255))
    status = Column(String(10))
    link = Column(String(255), nullable=False)
    created_at = Column(DateTime, default=datetime.now().strftime("%Y-%m-%d %H:%M:%S"))

File: backend_node/app/Models/Sequelize/Process.js
module.exports = (sequelize, DataTypes) => {

    const Process = sequelize.define(
        'processes',
        {
            id: {
                type: DataTypes.INTEGER,
                autoIncrement: true,
                primaryKey: true,
                allowNull: false,
            },
            numero_processo: {
                type: DataTypes.STRING(50),
                allowNull: true,
            },
            data_disponibilizacao: {
                type: DataTypes.DATE,
                allowNull: true,
            },
            autores: {
                type: DataTypes.STRING(255),
                allowNull: true,
            },
            advogados: {
                type: DataTypes.STRING(255),
                allowNull: true,
            },
            conteudo_publicacao: {
                type: DataTypes.TEXT('long'),
                allowNull: true,
            },
            valor_bruto: {
                type: DataTypes.STRING(50),
                allowNull: true,
            },
            valor_liquido: {
                type: DataTypes.STRING(10),
                allowNull: true,
            },
            juros_moratorios: {
                type: DataTypes.STRING(50),
                allowNull: true,
            },
            honorarios_advocaticios: {
                type: DataTypes.STRING(50),
                allowNull: true,
            },
            reu: {
                type: DataTypes.STRING(255),
                allowNull: true,
            },
            status: {
                type: DataTypes.STRING(10),
                allowNull: true,
            },
            link: {
                type: DataTypes.STRING(255),
                allowNull: true,
            },
            created_at: {
                type: DataTypes.DATE,
                allowNull: true,
            },
        },
        {
            createdAt: false,
            updatedAt: false,
            freezeTableName: true,
        },
    );

    Process.associate = (models) => {
        Process.hasMany(models.UserProcess, { foreignKey: 'process_id', as: 'user_processes' });
    };

    return Process;

};
